File: TaeJin/Ex07/app/src/main/java/org/techtown/ex07/MainActivity.java
package org.techtown.ex07;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }
    public void login(View view){
        Intent intent =new Intent(getApplicationContext(),menu.class);
        startActivityForResult(intent,100);

    }
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == 100 && resultCode == RESULT_OK) {
            Toast.makeText(getApplicationContext(), data.getStringExtra("res"), Toast.LENGTH_LONG).show();
        }


    }
}

File: RecyclerViewDemo1/app/src/main/java/org/techtown/recyclerviewdemo1/MainActivity.kt
package org.techtown.recyclerviewdemo1

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import org.techtown.recyclerviewdemo1.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
  private lateinit var binding: ActivityMainBinding
  val fruitsList =
    listOf(
      Fruit("Mango", "Tom"),
      Fruit("Apple", "Joe"),
      Fruit("Banana", "Mark"),
      Fruit("Guava", "Mike"),
      Fruit("Lemon", "sumin")
    )

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

    binding.myRecyclerView.setBackgroundColor(Color.YELLOW)
    binding.myRecyclerView.layoutManager = LinearLayoutManager(this)
    binding.myRecyclerView.adapter = MyRecyclerViewAdapter(fruitsList,
      { selectedFruitItem: Fruit -> listItemClicked(selectedFruitItem) })
  }

  private fun listItemClicked(fruit: Fruit) {
    Toast.makeText(this@MainActivity, "Suppier name is ${fruit.supplier}", Toast.LENGTH_LONG).show()
  }
}